ksql-engine/src/main/java/io/confluent/ksql/QueryEngine.java

```diff
@@ -268,15 +266,20 @@ private AbstractStreamCreateStatement maybeAddFieldsFromSchemaRegistry(
               ksqlTopic.getKsqlTopicSerDe().getSerDe().toString()
           )
       );
-      streamCreateStatement = streamCreateStatement.copyWith(
+      final AbstractStreamCreateStatement statementWithProperties = streamCreateStatement.copyWith(
           streamCreateStatement.getElements(),
-          newProperties
+          newProperties);
+      return StatementWithSchema.forStatement(
+          statementWithProperties,
+          SqlFormatter.formatSql(statementWithProperties),
```

Comment:
One thing to note for the future: the test coverage for `SqlFormatter` is very sparse. We rely heavily on this code to serialize SQL statements, and we should really expand coverage for this class at some point. 

One way to do this is to expand the query translation test to execute the re-serialized version of the statement in addition to the regular version to ensure that the result is the same. 

Does that make sense? If so, would you mind filing an issue with that suggestion. We can then fix it going forward as we focus on improving our test coverage.

Reply:
Filed https://github.com/confluentinc/ksql/issues/1571